sklearn/pipeline.py

```diff
@@ -657,6 +659,21 @@ def n_features_in_(self):
         # delegate to first step (which will call _check_is_fitted)
         return self.steps[0][1].n_features_in_
 
+    def __is_fitted__(self):
```

Comment:
Looking at this again, I think it is nicer to namespace this and pull it apart from Python "magic methods":

```suggestion
    def _sk_is_fitted_(self):
```

Reply:
I'm happy to namespace it. But I'm less sure about a magic method vs a private one. This is something which can be used by third party estimators. The question is, do we want to have it as a part of our "developer API" (which we don't really have yet), or really a private API which can change anytime.

Reply:
I would prefer the `__is_fitted__` but I don't have any argument in favor :)

Reply:
> . But I'm less sure about a magic method vs a private one.

I'm mostly thinking about double underscore or single underscore. double underscore == more "python magic method like".

> The question is, do we want to have it as a part of our "developer API" (which we don't really have yet), or really a private API which can change anytime.

I think `_sk_is_fitted_` is simple enough to make it a "developer API". I do not see it changing. (The only thing we need to decide on is the naming.)

Reply:
> double underscore == more "python magic method like".


That's not how I think about it though. For instance, I know if I write something which should work with numpy, I can implement `__array__`. But I almost never end up having to implement a private, i.e. single underscore, method to work with a different framework.

Reply:
Looking at all the libraries I am faimliar with. Most use double underscore. Numpy has all its `__arary_*__`, and dask has `__dask_*__`.

The only exception is jupyter using `_repr_html_` for their HTML display.

Reply:
> Looking at all the libraries I am faimliar with. Most use double underscore. Numpy has all its __array_*__, and dask has __dask_*__.

This seems to be a good argument to me to mimic.

Reply:
Ok, then I'll make it `__sk_is_fitted__`

Reply:
I think I had similar surprise to @thomasjpfan here. Ipython uses single underscore `_repr_html_` but perhaps that is the exception.

Reply:
Reopening this discussion: make we should expand `sk` to `sklearn` for namespace scoping as we are not the only scikit.